pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -273,116 +282,53 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance dynatr
 	}
 
 	if instance.GetOneAgentStatus().Version == "" {
-		desired, err := dtc.GetLatestAgentVersion(dtclient.OsUnix, dtclient.InstallerTypeDefault)
-		if err != nil {
-			return false, fmt.Errorf("failed to get desired version: %w", err)
+		if instance.GetOneAgentSpec().UseImmutableImage {
```

Comment:
This is only relevant if the customer doesn't use a custom image, otherwise the version is whatever is inside the image (can we know the version there?)

Reply:
Added a check for an empty string as image to make sure we do not have a custom image here